libs/core/likwid/include/hpx/likwid/likwid_executor.hpp

```diff
@@ -0,0 +1,230 @@
+        {
+            hpx::parallel::execution::post(exec.exec_,
+                hook_wrapper<F>{exec, std::forward<F>(f)},
+                std::forward<Ts>(ts)...);
+        }
+
+        // BulkOneWayExecutor interface
+        template <typename F, typename S, typename... Ts>
+        friend decltype(auto) tag_invoke(hpx::parallel::execution::bulk_sync_execute_t,
+            likwid_executor const& exec, F&& f, S const& shape, Ts&&... ts)
+        {
+            return hpx::parallel::execution::bulk_sync_execute(exec.exec_,
+                hook_wrapper<F>{exec, std::forward<F>(f)}, shape,
+                std::forward<Ts>(ts)...);
+        }
+
+        // BulkTwoWayExecutor interface
+        template <typename F, typename S, typename... Ts>
+        friend decltype(auto) tag_invoke(hpx::parallel::execution::bulk_async_execute_t,
+            likwid_executor const& exec, F&& f, S const& shape, Ts&&... ts)
+        {
+            return hpx::parallel::execution::bulk_async_execute(exec.exec_,
+                hook_wrapper<F>{exec, std::forward<F>(f)}, shape,
+                std::forward<Ts>(ts)...);
+        }
+
+        template <typename F, typename S, typename Future, typename... Ts>
+        friend decltype(auto) tag_invoke(hpx::parallel::execution::bulk_then_execute_t,
+            likwid_executor const& exec, F&& f, S const& shape, Future&& predecessor, Ts&&... ts)
+        {
+            return hpx::parallel::execution::bulk_then_execute(exec.exec_,
+                hook_wrapper<F>{exec, std::forward<F>(f)}, shape,
+                std::forward<Future>(predecessor), std::forward<Ts>(ts)...);
+        }
+
+        // support all properties exposed by the wrapped executor
+        // clang-format off
+        template <typename Tag, typename Property,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::execution::experimental::is_scheduling_property_v<Tag> &&
+                hpx::functional::is_tag_invocable_v<Tag, BaseExecutor, Property>
+            )>
+        // clang-format on
+        friend likwid_executor tag_invoke(
+            Tag tag, likwid_executor const& exec, Property&& prop)
+        {
+            return likwid_executor(hpx::functional::tag_invoke(
+                tag, exec.exec_, HPX_FORWARD(Property, prop)));
+        }
+
+        // clang-format off
+        template <typename Tag,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::execution::experimental::is_scheduling_property_v<Tag> &&
+                hpx::functional::is_tag_invocable_v<Tag, BaseExecutor>
+            )>
+        // clang-format on
+        friend decltype(auto) tag_invoke(
+            Tag tag, likwid_executor const& exec)
+        {
+            return hpx::functional::tag_invoke(tag, exec.exec_);
+        }
+
+    private:
+        using thread_hook = hpx::function<void()>;
+
+        BaseExecutor& exec_;
+        thread_hook on_start_;
+        thread_hook on_stop_;
+    };
+
```

Comment:
Would you mind adding a CTAD deduction guide that would possibly simplify code? That might even make obsolete the `make_likwid_executor` helper below.